Assets/Scripts/SS3D/Systems/Health/BodyParts/Bodypart.cs

```diff
@@ -415,6 +403,14 @@ public bool TryInflictDamage(BodyLayerType type, DamageTypeQuantity damageTypeQu
 
         BodyLayer layer = FirstBodyLayerOfType(type);
         if (!BodyLayers.Contains(layer)) return false;
+        InflictDamage(layer, damageTypeQuantity);
+
+        OnDamageInflicted?.Invoke(this, EventArgs.Empty);
+        return true;    
+    }
+
+    protected virtual void InflictDamage(BodyLayer layer, DamageTypeQuantity damageTypeQuantity)
```

Comment:
Github change comparisons is unreadable for me there .... I'm not sure what's going on despite having the code open, can you explain the changes ?

Reply:
I've extracted InflictDamage code from TryInflictDamage to override it after in child classes